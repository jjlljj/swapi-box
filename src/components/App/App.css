.App {
  text-align: center;
  background: blue;

}
