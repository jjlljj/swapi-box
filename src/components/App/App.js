import React, { Component } from 'react';
import './App.css';
import Main from '../Main/Main';
import Header from '../Header/Header'
import { getScroll, getPeople, getPlanets, getVehicles } from '../../dataHelper/dataHelper.js'
import Welcome from '../Welcome/Welcome'
import Card from '../Card/Card'

class App extends Component {
  constructor() {
    super();
    this.state = {
      openingText: {},
      people: null,
      planets: null,
      vehicles: null
    };
  }

  async componentDidMount() {
    //const openingText = await getScroll()

    let { people, planets, vehicles } = this.lastFromSto()

    people = people || await getPeople()
    planets = planets || await getPlanets()
    vehicles = vehicles || await getVehicles()
    
    // this.dataToSto({ people, planets, vehicles })
    this.setState({ people, planets, vehicles })

  }

  dataToSto({ people, planets, vehicles }) {
    const toSet = JSON.stringify({ people, planets, vehicles })
    localStorage.setItem('swapiData', toSet)
  }

  lastFromSto() {
    const swapiData = localStorage.getItem('swapiData')
    const { people, planets, vehicles } = JSON.parse(swapiData) || this.state 

    return { people, planets, vehicles }
  }

  render() {
    return (
      <div className="App">
        <Header />
        { this.state.planets &&
        <Card card={this.state.planets[0]} />
        }
        <Main />
      </div>
    );
  }
}
        //<Welcome {...this.state.openingText}/>

export default App;
