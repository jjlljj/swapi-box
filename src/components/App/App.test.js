/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

const localStorage = {}

describe('App', () => {

  it('should exist', () => {
    const renderedComponent = shallow(<App />, {disableLifecycleMethods: true});

    expect(false).toEqual(true);
  });
  
});
