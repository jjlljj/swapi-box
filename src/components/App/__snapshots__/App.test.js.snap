// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App state should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Main
          addToFav={[Function]}
          favorites={Array []}
          fetchPeople={[Function]}
          fetchPlanets={[Function]}
          fetchVehicles={[Function]}
          openingText={Object {}}
          people={null}
          planets={null}
          vehicles={null}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addToFav": [Function],
          "favorites": Array [],
          "fetchPeople": [Function],
          "fetchPlanets": [Function],
          "fetchVehicles": [Function],
          "openingText": Object {},
          "people": null,
          "planets": null,
          "vehicles": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Main
            addToFav={[Function]}
            favorites={Array []}
            fetchPeople={[Function]}
            fetchPlanets={[Function]}
            fetchVehicles={[Function]}
            openingText={Object {}}
            people={null}
            planets={null}
            vehicles={null}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addToFav": [Function],
            "favorites": Array [],
            "fetchPeople": [Function],
            "fetchPlanets": [Function],
            "fetchVehicles": [Function],
            "openingText": Object {},
            "people": null,
            "planets": null,
            "vehicles": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
